File: Backend/src/main/java/com/app/distrifoods/app/entities/Pedido.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.app.distrifoods.app.entities;

import java.io.Serializable;
import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name="pedidos")
public class Pedido implements Serializable  {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY) 
    @Column(name="id_pedido")
    private Integer id;
    @Column(name="id_cliente", nullable=false)
    private Integer id_cliente;
    @Column(name="id_met_pago", nullable=false)
    private Integer id_metodo;
    @Column(nullable=false)
    private Date fecha;
    @Column(name="id_estado", nullable=false)
    private Integer id_estado;
    @Column(nullable=false)
    private Float precio;
}

File: Frontend/rotonda-front/src/app/models/pedido.ts
import { ClienteCompleto } from "./cliente";
import { Estado } from "./estado";
import { MetodoPago } from "./metodoPago";
import { ProductoCompleto } from "./producto";

export class Pedido {
  idCliente:  number;
  idMetodo: number;
  fecha: Date;
  idEstado: number;
  precio: number;
  productos: ProductoCompleto[];

  constructor(idCliente:  number, idMetodo: number, fecha: Date, idEstado: number, precio: number, productos: ProductoCompleto[]){
    this.idCliente = idCliente;
    this.idMetodo = idMetodo;
    this.fecha = fecha;
    this.idEstado = idEstado;
    this.precio = precio;
    this.productos = productos;
  }
}


export interface PedidoCompleto {
  id: number;
  cliente: ClienteCompleto;
  metodoPago: MetodoPago;
  fecha: Date;
  estado: Estado;
  precio: number;
  productos: ProductoCompleto[];
}
